packages/fluence-js/src/internal/compilerSupport/v3impl/callFunction.ts

```diff
@@ -124,9 +138,25 @@ const extractArgs = (
         config = args[numberOfExpectedArgs];
     }
 
+    if (structuredArgs.length !== numberOfExpectedArgs) {
+        throw new Error(`Incorrect number of arguments. Expecting ${numberOfExpectedArgs}`);
+    }
+
+    const argsRes = Object.entries(argumentTypes)
+        .map(([name, _], ix) => ({ [name]: structuredArgs[ix] }))
+        .reduce((acc, kp) => ({ ...acc, ...kp }), {});
+
     return {
         peer: peer,
         config: config,
-        args: structuredArgs,
+        args: argsRes,
     };
 };
+
+const getArgumentTypes = (def: FunctionCallDef) => {
+    if (def.arrow.domain.tag !== 'labeledProduct') {
+        throw new Error('Should be impossible');
```

Comment:
If it should be impossible - maybe no need for this chek? Or if you think that it is possible - maybe a better error message for the user so he knows what he can do if this situation accures

Reply:
I see now. I understand that if this happens - compiler output is wrong right?

Reply:
Correct, the compiler output would be wrong.

Reply:
Maybe would be still not that bad to say it in error message